presentation/src/main/java/com/ragdroid/mvi/main/MainActivityFragment.kt

```diff
@@ -0,0 +1,80 @@
+package com.ragdroid.mvi.main
+
+import android.os.Bundle
+import android.support.v7.widget.LinearLayoutManager
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.fueled.reclaim.ItemsViewAdapter
+import com.jakewharton.rxbinding2.support.v4.widget.refreshes
+import com.ragdroid.mvi.R
+import com.ragdroid.mvi.databinding.FragmentMainBinding
+import com.ragdroid.mvi.helpers.BindFragment
+import com.ragdroid.mvi.items.CharacterItem
+import com.ragdroid.mvi.models.CharacterModel
+import dagger.android.support.DaggerFragment
+import io.reactivex.Observable
+import javax.inject.Inject
+
+/**
+ * A placeholder fragment containing a simple view.
+ */
+class MainActivityFragment : DaggerFragment(), MainFragmentView {
+
+    lateinit @Inject var presenter: MainFragmentPresenter
+    //delegate the binding initialization to BindFragment delegate
+    private val binding: FragmentMainBinding by BindFragment(R.layout.fragment_main)
+    private lateinit var adapter: ItemsViewAdapter
```

Comment:
You can make it `val` and lazy initialise like `private val adapter by lazy { ItemsViewAdapter(context) }`

Reply:
Good One 👍 Thanks